### PR TITLE
printing non grouped statements and wadler helpers

### PR DESCRIPTION
**Pull request by Amine-Mike**

### CODE REVIEW

File: lib/wadler/print.rb [hunk 1]
```diff
@@ -132,5 +132,70 @@ def breakable(str = ' ', width: str.length, line_continuation: '')
     def break(line_continuation: '')
       tokens << Oppen.line_break(line_continuation:, offset: current_indent)
     end
+
+    ## Helpers ##
```
Comment (stackmystack) [inline lib/wadler/print.rb]: There's a YARD "group" thing, use it.

File: lib/wadler/print.rb [hunk 2]
```diff
@@ -128,5 +132,70 @@ def breakable(str = ' ', width: str.length, line_continuation: '')
...
+
+    # Set a base indenetaion level to the printer.
+    #
+    # @param indent [Integer]
+    #
+    # @return [Nil]
+    def set_base_indent(indent = 0)
```
Comment (stackmystack) [inline lib/wadler/print.rb]:
```suggestion
    def base_indent(indent = 0)
```

File: test/wadler_test.rb [hunk 3]
```diff
@@ -610,4 +610,40 @@ def altshow(out)
...
     end
   end
+
+  describe 'must not break if no group was given' do
+    it 'must work with only a string' do
+      out = Oppen::Wadler.new
+      out.text 'Hello World'
+      _(out.output).must_equal 'Hello World'
+    end
+
```
Comment (stackmystack) [inline test/wadler_test.rb]:
```suggestion
  describe 'break if no group was given' do
    it 'works with a string only' do
      out = Oppen::Wadler.new
      out.text 'Hello World'
      _(out.output).must_equal 'Hello World'
    end

```

Let's use this phrasing from now on for tests. This is what minitest does anyway.
I will add a ticket to refactor the remainder of tests (outside this PR).

File: test/wadler_test.rb [hunk 4]
```diff
@@ -610,4 +610,40 @@ def altshow(out)
...
+      out = Oppen::Wadler.new
+      out.breakable ''
+      _(out.output).must_equal ''
+    end
+
+    it 'must work with only a nest' do
+      out = Oppen::Wadler.new
+      out.nest(0) {
+        out.text 'Hello World'
+      }
```
Comment (stackmystack) [inline test/wadler_test.rb]:
```suggestion
      out.nest(0) { out.text 'Hello World' }
```

File: test/wadler_test.rb [hunk 5]
```diff
@@ -610,4 +610,40 @@ def altshow(out)
...
+      _(out.output).must_equal 'Hello World'
+    end
+
+    it 'must work with multiple strings and breakables' do
+      out = Oppen::Wadler.new(width: 10)
+      out.text 'Hello World'
+      out.breakable
+      out.text 'How are you doig World'
+      out.breakable
+      out.text 'GoodBye World'
```
Comment (stackmystack) [inline test/wadler_test.rb:639]:
```suggestion
      out
        .text('Hello World')
        .breakable
        .text('How are you doig World')
        .breakable
        .text('GoodBye World')
```

Use this style everywhere.

**Review by stackmystack — CHANGES_REQUESTED** [review]

File: test/wadler_test.rb (hunk 5 above)
Comment (Amine-Mike) [reply test/wadler_test.rb:639]: This does not work. all Wadler methods do not return self.

Comment (stackmystack) [reply test/wadler_test.rb:639]: We should. In a different PR. And we need to change all tests to support it.

We also want `wrap/when` in here, and also `do`.

But let's not bother with it now. I will open a ticket.

**Review by stackmystack — APPROVED** [review]

**Pull request merged**